File: minecraft-web-manager/old-nodejs-system/backend/src/api/controllers/server.controller.js
import ServerService from '../../services/server.service.js';
import AuditLog from '../../models/AuditLog.js';

const serverService = new ServerService();

class ServerController {
  /**
   * GET /api/server/status
   * Obtener estado del servidor
   */
  static async getStatus(req, res) {
    try {
      const status = await serverService.getStatus();
      res.json({
        success: true,
        data: status
      });
    } catch (error) {
      console.error('Error al obtener estado:', error);
      res.status(500).json({
        success: false,
        message: error.message
      });
    }
  }

  /**
   * POST /api/server/start
   * Iniciar servidor
   */
  static async start(req, res) {
    try {
      const result = await serverService.start();

      // Registrar en audit log
      AuditLog.logServerAction(
        req.user.userId,
        'server_start',
        { message: result.message },
        req.ip
      );

      // Emitir evento via Socket.IO
      if (req.app.get('io')) {
        req.app.get('io').emit('server-status', { action: 'started' });
      }

      res.json({
        success: true,
        message: result.message
      });
    } catch (error) {
      console.error('Error al iniciar servidor:', error);
      res.status(500).json({
        success: false,
        message: error.message
      });
    }
  }

  /**
   * POST /api/server/stop
   * Detener servidor
   */
  static async stop(req, res) {
    try {
      const result = await serverService.stop();

      // Registrar en audit log
      AuditLog.logServerAction(
        req.user.userId,
        'server_stop',
        { message: result.message },
        req.ip
      );

      // Emitir evento via Socket.IO
      if (req.app.get('io')) {
        req.app.get('io').emit('server-status', { action: 'stopped' });
      }

      res.json({
        success: true,
        message: result.message
      });
    } catch (error) {
      console.error('Error al detener servidor:', error);
      res.status(500).json({
        success: false,
        message: error.message
      });
    }
  }

  /**
   * POST /api/server/restart
   * Reiniciar servidor
   */
  static async restart(req, res) {
    try {
      const result = await serverService.restart();

      // Registrar en audit log
      AuditLog.logServerAction(
        req.user.userId,
        'server_restart',
        { message: result.message },
        req.ip
      );

      // Emitir evento via Socket.IO
      if (req.app.get('io')) {
        req.app.get('io').emit('server-status', { action: 'restarted' });
      }

      res.json({
        success: true,
        message: result.message
      });
    } catch (error) {
      console.error('Error al reiniciar servidor:', error);
      res.status(500).json({
        success: false,
        message: error.message
      });
    }
  }

  /**
   * GET /api/server/logs
   * Obtener logs del servidor
   */
  static async getLogs(req, res) {
    try {
      const lines = parseInt(req.query.lines) || 100;
      const logs = await serverService.getLogs(lines);

      res.json({
        success: true,
        data: logs
      });
    } catch (error) {
      console.error('Error al obtener logs:', error);
      res.status(500).json({
        success: false,
        message: error.message
      });
    }
  }

  /**
   * GET /api/server/info
   * Obtener información del servidor
   */
  static async getInfo(req, res) {
    try {
      const info = await serverService.getInfo();

      res.json({
        success: true,
        data: info
      });
    } catch (error) {
      console.error('Error al obtener información:', error);
      res.status(500).json({
        success: false,
        message: error.message
      });
    }
  }

  /**
   * POST /api/server/command
   * Enviar comando RCON
   */
  static async sendCommand(req, res) {
    try {
      const { command } = req.body;

      if (!command) {
        return res.status(400).json({
          success: false,
          message: 'Comando requerido'
        });
      }

      const result = await serverService.sendCommand(command);

      // Registrar en audit log
      AuditLog.logServerAction(
        req.user.userId,
        'rcon_command',
        { command, output: result.output },
        req.ip
      );

      res.json({
        success: true,
        data: result
      });
    } catch (error) {
      console.error('Error al enviar comando:', error);
      res.status(500).json({
        success: false,
        message: error.message
      });
    }
  }
}

export default ServerController;

File: minecraft-web-manager/backend-python/app/api/controllers/server_controller.py
from fastapi import HTTPException, status
from typing import Optional

from app.services.server_service import ServerService
from app.services.bash_service import BashService


class ServerController:
    def __init__(self):
        self.server_service = ServerService()
        self.bash_service = BashService()
    
    async def get_status(self):
        """Obtener estado actual del servidor"""
        try:
            status_data = await self.server_service.get_status()
            return status_data
        except Exception as e:
            raise HTTPException(
                status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
                detail=f"Error al obtener estado: {str(e)}"
            )
    
    async def get_info(self):
        """Obtener información del servidor"""
        try:
            info = await self.server_service.get_info()
            return info
        except Exception as e:
            raise HTTPException(
                status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
                detail=f"Error al obtener información: {str(e)}"
            )
    
    async def start(self):
        """Iniciar servidor"""
        try:
            # Verificar que no esté corriendo
            status_data = await self.server_service.get_status()
            if status_data.get("running"):
                raise HTTPException(
                    status_code=status.HTTP_400_BAD_REQUEST,
                    detail="El servidor ya está corriendo"
                )
            
            result = await self.server_service.start()
            return {"message": "Servidor iniciado correctamente", "result": result}
        except HTTPException:
            raise
        except Exception as e:
            raise HTTPException(
                status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
                detail=f"Error al iniciar servidor: {str(e)}"
            )
    
    async def stop(self):
        """Detener servidor"""
        try:
            # Verificar que esté corriendo
            status_data = await self.server_service.get_status()
            if not status_data.get("running"):
                raise HTTPException(
                    status_code=status.HTTP_400_BAD_REQUEST,
                    detail="El servidor no está corriendo"
                )
            
            result = await self.server_service.stop()
            return {"message": "Servidor detenido correctamente", "result": result}
        except HTTPException:
            raise
        except Exception as e:
            raise HTTPException(
                status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
                detail=f"Error al detener servidor: {str(e)}"
            )
    
    async def restart(self):
        """Reiniciar servidor"""
        try:
            result = await self.server_service.restart()
            return {"message": "Servidor reiniciado correctamente", "result": result}
        except Exception as e:
            raise HTTPException(
                status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
                detail=f"Error al reiniciar servidor: {str(e)}"
            )
    
    async def get_logs(self, lines: int = 100):
        """Obtener logs del servidor"""
        try:
            logs = await self.server_service.get_logs(lines)
            return {"logs": logs, "lines": len(logs)}
        except Exception as e:
            raise HTTPException(
                status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
                detail=f"Error al obtener logs: {str(e)}"
            )
    
    async def send_command(self, command: str):
        """Enviar comando RCON al servidor"""
        try:
            # Verificar que el servidor esté corriendo
            status_data = await self.server_service.get_status()
            if not status_data.get("running"):
                raise HTTPException(
                    status_code=status.HTTP_400_BAD_REQUEST,
                    detail="El servidor no está corriendo"
                )
            
            result = await self.bash_service.rcon_command(command)
            return {
                "command": command,
                "output": result.get("stdout", ""),
                "success": result.get("success", False)
            }
        except HTTPException:
            raise
        except Exception as e:
            raise HTTPException(
                status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
                detail=f"Error al enviar comando: {str(e)}"
            )
